packages/nimble-components/src/wafer-map/modules/hover-handler.ts

```diff
@@ -45,36 +45,53 @@ export class HoverHandler {
             : Math.ceil;
         const yRoundFunction = originLocation === WaferMapOriginLocation.bottomLeft
             || originLocation === WaferMapOriginLocation.bottomRight
-            ? Math.floor
-            : Math.ceil;
+            ? Math.ceil
+            : Math.floor;
         // go to x and y scale to get the x,y values of the die.
         const x = xRoundFunction(
-            wafermap.dataManager.invertedHorizontalScale(
-                mousePosition.x - wafermap.dataManager.margin.left
-            )
+            wafermap.matrixRenderer.invertedHorizontalScale.a
+                + wafermap.matrixRenderer.invertedHorizontalScale.b
+                    * (mousePosition.x - wafermap.matrixRenderer.margin.left)
         );
         const y = yRoundFunction(
-            wafermap.dataManager.invertedVerticalScale(
-                mousePosition.y - wafermap.dataManager.margin.top
-            )
+            wafermap.matrixRenderer.invertedVerticalScale.a
+                + wafermap.matrixRenderer.invertedVerticalScale.b
+                    * (mousePosition.y - wafermap.matrixRenderer.margin.top)
         );
         return { x, y };
     }
 
-    private hoversOverDie(
-        wafermap: WaferMap,
-        mousePosition: PointCoordinates
-    ): boolean {
-        const rgba = wafermap.canvasContext.getImageData(
-            mousePosition.x,
-            mousePosition.y,
-            1,
-            1
-        ).data;
-        let rgbaSum = 0;
-        for (const color of rgba) {
-            rgbaSum += color;
-        }
-        return rgbaSum > 0;
+    private getWaferMapDie(
+        _dieCoordinates: PointCoordinates
+    ): WaferMapDie | undefined {
+        // const colIndex = this.wafermap.dieMatrix.dieColIndexArray.indexOf(
```

Comment:
@rajsite In this place the handler was searching for the specific x and y coordinate and returning the value and metadata. the arrow table does not support filtering or any kind of search without iterating over the full table [as mentioned here](https://github.com/apache/arrow/issues/13233). a solution can be this [library on top of arrow](https://uwdata.github.io/arquero/api/verbs#filter) like you mentioned. I will test these two options and compare them to see if they have different performances

Reply:
Yea the hope is that these higher-level apis give easier to maintain code / algorithms while having good performance. 🤞